src-tauri/tests/tests/credential_offer.rs

```diff
@@ -0,0 +1,357 @@
+use wiremock::{Mock, MockServer, ResponseTemplate};
+
+#[tokio::test]
+async fn download_credential_logo() {
+    *ASSETS_DIR.lock().unwrap() = TempDir::new().unwrap().into_path();
+
+    let mock_server = MockServer::start().await;
+    Mock::given(method("GET"))
+        .and(path("/offer/1"))
+        .respond_with(
+            ResponseTemplate::new(200).set_body_json(CredentialOffer::<CredentialFormats> {
+                credential_issuer: format!("{}", &mock_server.uri()).parse().unwrap(),
+                credentials: vec![CredentialsObject::ByReference("UniversityDegreeCredential".to_string())],
+                grants: None,
+            }),
+        )
+        .expect(1)
+        .mount(&mock_server)
+        .await;
+
+    Mock::given(method("GET"))
+        .and(path("/.well-known/openid-credential-issuer"))
+        .respond_with(
+            ResponseTemplate::new(200).set_body_json(CredentialIssuerMetadata::<CredentialFormats> {
+                credential_endpoint: Url::parse("https://server.example.com/credential").unwrap(),
+                credentials_supported: vec![CredentialsSupportedObject {
+                    id: None,
+                    credential_format: CredentialFormats::JwtVcJson(Parameters {
+                        format: JwtVcJson,
+                        parameters: (
+                            jwt_vc_json::CredentialDefinition {
+                                type_: vec![
+                                    "VerifiableCredential".to_string(),
+                                    "UniversityDegreeCredential".to_string(),
+                                ],
+                                credential_subject: Some(json!({
+                                    "given_name": {
+                                        "display": [
+                                            {
+                                                "name": "Given Name",
+                                                "locale": "en-US"
+                                            }
+                                        ]
+                                    },
+                                    "family_name": {
+                                        "display": [
+                                            {
+                                                "name": "Surname",
+                                                "locale": "en-US"
+                                            }
+                                        ]
+                                    },
+                                    "degree": {},
+                                    "gpa": {
+                                        "display": [
+                                            {
+                                                "name": "GPA"
+                                            }
+                                        ]
+                                    }
+                                })),
```

Comment:
the occurrences of `credential_subject` in these test files are optional and not relevant for what's being tested, so I would suggest to set the values to `None` in order to make the tests a bit less bloated.